code/modules/awaymissions/maploader/reader.dm

```diff
@@ -41,8 +41,8 @@ var/global/dmm_suite/preloader/_preloader = new
 	var/list/grid_models = list()
 	var/key_len = 0
 
-	dmmRegex.next = 1
-	while(dmmRegex.Find(tfile, dmmRegex.next))
+	var/regex_index = 1
+	while(dmmRegex.Find(tfile, regex_index))
```

Comment:
Yeah, shouldn't you be setting the value of `regex_index` to `dmmRegex.next` first thing in the while loop